x-pack/plugins/ingest_manager/public/applications/ingest_manager/sections/agent_config/components/actions_menu.tsx

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { memo, useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiContextMenuItem, EuiPortal } from '@elastic/eui';
+import { useCapabilities, useLink } from '../../../hooks';
+import { ContextMenuActions } from '../../../components';
+import { ConfigYamlFlyout } from './config_yaml_flyout';
+
+export const AgentConfigActionMenu = memo<{ configId: string; fullButton?: boolean }>(
+  ({ configId, fullButton = false }) => {
+    const { getHref } = useLink();
+    const hasWriteCapabilities = useCapabilities().write;
+    const [isYamlFlyoutOpen, setIsYamlFlyoutOpen] = useState<boolean>(false);
+    return (
+      <>
+        {isYamlFlyoutOpen ? (
+          <EuiPortal>
+            <ConfigYamlFlyout configId={configId} onClose={() => setIsYamlFlyoutOpen(false)} />
+          </EuiPortal>
+        ) : null}
+        <ContextMenuActions
+          button={
+            fullButton
+              ? {
+                  props: {
+                    iconType: 'arrowDown',
+                    iconSide: 'right',
+                  },
+                  children: (
+                    <FormattedMessage
+                      id="xpack.ingestManager.agentConfigActionMenu.buttonText"
+                      defaultMessage="Actions"
+                    />
+                  ),
+                }
+              : undefined
+          }
+          items={[
+            <EuiContextMenuItem
```

Comment:
Would you be ok if we had a URL param in the route that would trigger this flyout? Does not have to be part of this PR.
One of the integration tests I have been thinking about for Endpoint "policy" data is to ensure that data (when created/edited) lands in the right place in the YAML file, which should give us an assurance that it will make it to the Endpoint exe as intended. Having a route to get this panel would make it easier to build up that integration test. 

Like I said - does not have to be part of this PR, but was wondering if you would be ok with such a change in the future when we get around to implementing that integration test 😃

Reply:
Yes, I'm ok with any url params to trigger any of our flyouts :)